File: go/raindrops/raindrops.go
package raindrops

import "strconv"

const testVersion = 2

func Convert(n int) string {
	var resp string
	if n == 1 {
		return "1"
	}
	if n%3 == 0 {
		resp += "Pling"
	}
	if n%5 == 0 {
		resp += "Plang"
	}
	if n%7 == 0 {
		resp += "Plong"
	}
	if resp == "" {
		resp = string(strconv.Itoa(n))
	}
	return resp
}

File: erlang/raindrops/src/raindrops.erl
-module(raindrops).

-export([convert/1]).


convert(Number) when is_integer(Number) ->
    case pling(Number rem 3)++plang(Number rem 5)++plong(Number rem 7) of
        "" ->
            integer_to_list(Number);
        RainSpeak ->
            RainSpeak
    end.

pling(0) ->
    "Pling";
pling(_) ->
    "".

plang(0) ->
    "Plang";
plang(_) ->
    "".

plong(0) ->
    "Plong";
plong(_) ->
    "".
